src/include/connection.h

```diff
@@ -151,7 +151,17 @@ struct __wt_bucket_storage {
  *  Heuristic controls configuration.
  */
 struct __wt_heuristic_controls {
-    uint32_t obsolete_tw_pages_dirty;
+    /*
+     * Number of btrees processed in the current checkpoint due to obsolete time window information.
+     */
+    wt_shared uint32_t obsolete_tw_btree_count;
+    /* Maximum number of btrees that can be processed due to obsolete time window information. */
+    wt_shared uint32_t obsolete_tw_btree_max;
```

Comment:
Does the max and dirty max also need to be shared? @ajmorton How do we handle configurable values?

Reply:
TSan is going to warn about any and every concurrent read/write. If the value can be reconfigured we'll need to treat them as atomic. 
As a rule of thumb I'd be happy to trust TSan testing on this. If it doesn't fail a test we can always come back to it when we add more TSan testing

Reply:
I'll try to remove the configuration parameters and keep `obsolete_tw_btree_count` as shared.

Reply:
Green [TSAN build](https://spruce.mongodb.com/version/66909c4f7865cb000799d72b/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC) after removing the wt_shared property from the connection cfg items.